test/ci_workflow_gen.go

```diff
@@ -213,12 +219,74 @@ type clusterTest struct {
 }
 
 type vitessTesterTest struct {
-	FileName         string
-	Name             string
-	RunsOn           string
-	GoPrivate        string
-	GoJunitReportSHA string
-	Path             string
+	*GitMetas
+	FileName  string
+	Name      string
+	RunsOn    string
+	GoPrivate string
+	Path      string
+}
+
+// getGitRefSHA fetches the HEAD SHA of a git repo + branch using the "git" command.
+func getGitRefSHA(ctx context.Context, url, branchOrTag string) (string, error) {
+	var stdout bytes.Buffer
+	cmd := exec.CommandContext(ctx, "git", "ls-remote", url, branchOrTag)
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
+		return "", err
+	}
+
+	// 'git ls-remote <url> <branchOrTag>' returns two text columns: a commit
+	// SHA and a reference. We expect the stdout to be a single line, which
+	// should always be true but we still validate.
+	// Example:
+	//    $ git ls-remote https://github.com/vitessio/go-junit-report HEAD
+	//    99fa7f0daf16db969f54a49139a14471e633e6e8	HEAD
+	for line := range strings.Lines(stdout.String()) {
+		fields := strings.Fields(line)
+		if len(fields) != 2 {
+			continue
+		}
+		if strings.Contains(fields[1], branchOrTag) {
+			return fields[0], nil
```

Comment:
Probably worth confirming that it’s a valid sha256 hex value (64 byte Len consisting of the limited charset).